client/asserts.go

```diff
@@ -27,6 +27,8 @@ import (
 	"net/url"
 	"strconv"
 
+	"golang.org/x/xerrors"
```

Comment:
we have a bit of a question of how much change we do to source files that start importing xerrors:

* do we turn all fmt.Errorf and errors.New into the respective xerrors calls?
* when do we switch to use %w: in general that is mostly useful if the error printed could be a network/store/snapd daemon error

Reply:
I have no strong opinion but my approach would be to not change move than we need for this PR. The errors faq in the wiki https://github.com/golang/go/wiki/ErrorValueFAQ warns that converting to %w should be done with care because all the error details we expose are now part of the API.

Reply:
As I said I'm not proposing to use %w everywhere but with this PR the codebase here is already inconsistent:

compare

https://github.com/snapcore/snapd/blob/96d0b2506f06ffac5be335ef265399cc7eaa5d45/client/asserts.go#L56

vs your new code:

https://github.com/snapcore/snapd/blob/96d0b2506f06ffac5be335ef265399cc7eaa5d45/client/asserts.go#L93

My first point stands.

Reply:
Right, sorry that I'm a bit slow. I changed this particular part because we need to get the error type (ConnectionError) from client.Known() to be able to distinguish it later in cmd_known.go. This is the entire motivation for this change. I did look over the fmt.Errorf() in client/asserts.go and it seems to make sense to change the error in AssertionTypes too. The others look ok. I can do this. I could also invent a new "type KnownError struct { oldErr error }" which would allow us to move forward without xerrors and the open questions. Any of these options works for me. Happy about suggestions!

Reply:
Thanks for the suggestion - after we talked a bit in a HO I think I know what to do and pushed https://github.com/snapcore/snapd/pull/7645